core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java

```diff
@@ -184,41 +154,39 @@ public void commitBlockTest() throws Exception {
     when(blockStoreMeta.getUsedBytesOnTiers()).thenReturn(usedBytesOnTiers);
 
     mBlockWorker.commitBlock(sessionId, blockId);
-    verify(mBlockMasterClient).commitBlock(mWorkerId, usedBytes,
-        tierAlias, blockId, length);
+    verify(mBlockMasterClient).commitBlock(anyLong(), eq(usedBytes), eq(tierAlias), eq(blockId),
+        eq(length));
     verify(mBlockStore).unlockBlock(lockId);
   }
 
   /**
-   * Tests commitBlock, the master-side RPC failed for the first time, expecting retry to succeed.
+   * Tests that commitBlock doesn't throw an exception when {@link BlockAlreadyExistsException} gets
+   * thrown by the block store.
    */
   @Test
   public void commitBlockOnRetryTest() throws Exception {
-    mBlockWorker = new DefaultBlockWorker();
     long blockId = mRandom.nextLong();
+    long length = mRandom.nextLong();
+    long lockId = mRandom.nextLong();
     long sessionId = mRandom.nextLong();
+    long usedBytes = mRandom.nextLong();
     String tierAlias = "MEM";
-    long initialBytes = 1;
-
-    Whitebox.setInternalState(mBlockWorker, "mBlockMasterClient", mBlockMasterClient);
-    PowerMockito.mockStatic(WorkerIdRegistry.class);
-    when(WorkerIdRegistry.getWorkerId()).thenReturn(Long.valueOf(1));
-
-    PowerMockito.doThrow(new IOException("Server RPC failure")).when(mBlockMasterClient)
-        .commitBlock(anyLong(), anyLong(), anyString(), anyLong(), anyLong());
-
-    String blockPath = mBlockWorker.createBlock(sessionId, blockId, tierAlias, initialBytes);
-    LocalFileBlockWriter writer = new LocalFileBlockWriter(blockPath);
-    writer.close();
-    try {
-      mBlockWorker.commitBlock(sessionId, blockId);
-    } catch (IOException e) {
-      // expect an IOException thrown
-    }
-
-    // Let's retry commitBlock
-    PowerMockito.doNothing().when(mBlockMasterClient)
-        .commitBlock(anyLong(), anyLong(), anyString(), anyLong(), anyLong());
+    HashMap<String, Long> usedBytesOnTiers = new HashMap<>();
+    usedBytesOnTiers.put(tierAlias, usedBytes);
+    BlockMeta blockMeta = PowerMockito.mock(BlockMeta.class);
+    BlockStoreLocation blockStoreLocation = PowerMockito.mock(BlockStoreLocation.class);
+    BlockStoreMeta blockStoreMeta = PowerMockito.mock(BlockStoreMeta.class);
+
+    when(mBlockStore.lockBlock(sessionId, blockId)).thenReturn(lockId);
+    when(mBlockStore.getBlockMeta(sessionId, blockId, lockId)).thenReturn(blockMeta);
+    when(mBlockStore.getBlockStoreMeta()).thenReturn(blockStoreMeta);
+    when(blockMeta.getBlockLocation()).thenReturn(blockStoreLocation);
+    when(blockStoreLocation.tierAlias()).thenReturn(tierAlias);
+    when(blockMeta.getBlockSize()).thenReturn(length);
+    when(blockStoreMeta.getUsedBytesOnTiers()).thenReturn(usedBytesOnTiers);
```

Comment:
Yes, it's the same setup used by `commitBlockTest`. We need it because we ignore `BlockAlreadyExistsException` and execute the full body of `commitBlock`, which makes these calls.